### PR TITLE
🐛 Source Salesforce: filter 'null' byte(s) in HTTP responses

### LINKED ISSUE #8300: 🐛 Source Salesforce 0.1.7 fails with _csv.Error: line contains NUL
## Enviroment
- **Airbyte version**: 0.32.6-alpha
- **OS Version / Instance**: GCP/GKE
- **Deployment**: Kubernetes
- **Source Connector and version**: Salesforce 0.1.7
- **Destination Connector and version**: Bigquery 0.5.0 (or denormalized 0.1.10)
- **Severity**: Critical
- **Step where error happened**: Sync job

## Current Behavior
The saleforce incremental sync fails:
```
Traceback (most recent call last):
  File "/airbyte/integration_code/source_salesforce/source.py", line 85, in read
    internal_config=internal_config,
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 136, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 181, in _read_incremental
    for record_counter, record_data in enumerate(records, start=1):
  File "/airbyte/integration_code/source_salesforce/streams.py", line 249, in read_records
    for count, record in self.download_data(url=job_full_url):
  File "/airbyte/integration_code/source_salesforce/streams.py", line 213, in download_data
    for i, row in enumerate(csv_data):
_csv.Error: line contains NUL
```

## Expected Behavior
The salesforce sync should work

## Logs

[logs-451-1.txt](https://github.com/airbytehq/airbyte/files/7624316/logs-451-1.txt)

## Steps to Reproduce
1. just trying to sync

## Are you willing to submit a PR?
YES !

### PR DESCRIPTION
**Pull request by grubberr**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? If yes, please make sure to include it here and in any changelogs with the 🚨🚨 emoji
What is the end result perceived by the user?

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

File: .github/workflows/publish-command.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@ jobs:
           GCP_GSM_CREDENTIALS: ${{ secrets.GCP_GSM_CREDENTIALS }}
       - run: |
           echo "$SPEC_CACHE_SERVICE_ACCOUNT_KEY" > spec_cache_key_file.json && docker login -u airbytebot -p ${DOCKER_PASSWORD}
-          ./tools/integrations/manage.sh publish airbyte-integrations/${{ github.event.inputs.connector }} ${{ github.event.inputs.run-tests }} --publish_spec_to_cache
+          ./tools/integrations/manage.sh publish airbyte-integrations/${{ github.event.inputs.connector }} false --publish_spec_to_cache
```
Comment (sherifnada) [inline .github/workflows/publish-command.yml:79]: make sure not to merge this to master. FYI you can always toggle this option via `/publish ... run-tests=false`

Comment (grubberr) [reply .github/workflows/publish-command.yml:79]: yes of course I will revert 
thanks for `run-tests=false`

File: airbyte-config/init/src/main/resources/config/STANDARD_SOURCE_DEFINITION/b117307c-14b6-41aa-9422-947e34922962.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "sourceDefinitionId": "b117307c-14b6-41aa-9422-947e34922962",
   "name": "Salesforce",
   "dockerRepository": "airbyte/source-salesforce",
-  "dockerImageTag": "0.1.6",
+  "dockerImageTag": "0.1.9",
```
Comment (sherifnada) [inline airbyte-config/init/src/main/resources/config/STANDARD_SOURCE_DEFINITION/b117307c-14b6-41aa-9422-947e34922962.json:5]: it's no longer needed to update the JSON files

**Review by sherifnada — APPROVED** [review]

**Pull request merged**